components/service-waiter/cmd/component.go

```diff
@@ -127,8 +138,56 @@ func init() {
 	componentCmd.Flags().StringVar(&componentCmdOpt.namespace, "namespace", "", "The namespace of deployment")
 	componentCmd.Flags().StringVar(&componentCmdOpt.component, "component", "", "Component name of deployment")
 	componentCmd.Flags().StringVar(&componentCmdOpt.labels, "labels", "", "Labels of deployment")
+	componentCmd.Flags().DurationVar(&componentCmdOpt.featureFlagTimeout, "feature-flag-timeout", 3*time.Minute, "The maximum time to wait for feature flag")
 
 	_ = componentCmd.MarkFlagRequired("namespace")
 	_ = componentCmd.MarkFlagRequired("component")
 	_ = componentCmd.MarkFlagRequired("labels")
 }
+
+func startWaitFeatureFlag(ctx context.Context, timeout time.Duration) {
+	featureFlagCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	client := experiments.NewClient(experiments.WithDefaultClient(nil))
```

Comment:
We need default client to nil, since it will never be nil without it -> it will return a client `NewAlwaysReturningDefaultValueClient`. Then `self-hosted` Gitpod Instance may need to keep wait for it